---
title: "RNA-seq time course"
author: "Yi Xie"
date: "19/07/2018"
output: html_document
---

# Background
This experiment is an oncogene withdrawal study where leukaemias are driven by tet transactivator expression of the CBFb-MYH11 fusion (known as inv(16), which is withdrawn upon Dox treatment. In this experiment, we have 3 untreated samples vs 3 samples at both day1, day2 and day6. We got the basic analysis below:

1) MD plots of day1 vs untreated, day2 vs untreated and day6 vs untreated.
2) Lists of DE genes with individual RPKM.
3) MD plots highlighting Cbfb and Myh11.



# Alignment&Counting
```{r eval=FALSE}
setwd("/stornext/General/data/user_managed/grpu_mritchie_1/YiXie/reference/mm/")
library(Rsubread)
library(Mus.musculus)

#build index
buildindex('Mus_musculus.GRCm38', reference = "Mus_musculus.GRCm38.dna.primary_assembly.fa",
           memory = 12000)

#alignment
fastq.files <- list.files("/stornext/General/data/user_managed/grpu_mritchie_1/MarkMcKenzie/cbf-myh11", pattern = "fastq.gz$", full.names=T)
bam <- gsub("fastq.gz", "subread.bam",fastq.files)

align(index = "Mus_musculus.GRCm38", readfile1 = fastq.files,input_format="FASTQ", output_format="BAM", 
      output_file=bam, nthreads=8)

#counting&annotation
bam.files <- list.files(path = "./bamfiles", pattern = ".bam$", full.names = TRUE)
counts <- featureCounts(bam.files, annot.inbuilt = "mm10")
names(counts) #see slots in counts
dim(counts$counts) #see the number of genes

geneid <- rownames(counts$counts)
genes <- select(Mus.musculus, keys = geneid, columns = c("SYMBOL", "TXCHROM","TXSTART","TXEND","GENENAME"), keytype = "ENTREZID")
genes <- genes[!duplicated(genes$ENTREZID),]
save(counts,genes,file="counts.rda")
```



# Preprocessing
```{r echo=TRUE}
library(edgeR)
library(limma)
library(Glimma)
load("counts.rda")

#preprocessing
rpkmvalues <- rpkm(counts$counts, gene.length = counts$annotation$Length)

colnames(rpkmvalues) <- gsub("X.stornext.General.data.user_managed.grpu_mritchie_1.YiXie.reference.mm.bamfiles.","",colnames(rpkmvalues))
colnames(rpkmvalues) <- gsub("_S._R1.subread.bam","",colnames(rpkmvalues))
colnames(rpkmvalues) <- samplenames <- gsub("_S.._R1.subread.bam","",colnames(rpkmvalues))

genes <- cbind(genes, "Length"=counts$annotation$Length, rpkmvalues)

x <- DGEList(counts = counts$counts, genes=genes)

group <- as.factor(c("D1","D1","D1","D6","D6","D2","D2","D2","Untrt","Untrt","Untrt","D6"))
x$samples$group <- group
rownames(x$samples) <- gsub("X.stornext.General.data.user_managed.grpu_mritchie_1.YiXie.reference.mm.","",rownames(x$samples))
x$samples
```

```{r echo=TRUE}
##filter lowly expressed
keep.exprs <- rowSums(cpm(x)>1) >=3 #row=genes col=samples; genes express in at least 3 samples
x <- x[keep.exprs,,keep.lib.sizes=F]
dim(counts$counts)
dim(x)
```

```{r echo=TRUE}
##normalising
x <- calcNormFactors(x, method = "TMM")
x$samples$norm.factors
```

```{r echo=TRUE, fig.align=center, fig.height=5, fig.width=5, fig.align='center'}
library(RColorBrewer)
par(mfrow=c(1,1))
col.group <- group
levels(col.group) <- brewer.pal(nlevels(col.group),"Set1") #change groupID to colorID
col.group <- as.character(col.group)
lcpm <- cpm(x, log=T)

plotMDS(lcpm, labels=group, main="MDS: Sample Groups", col=col.group)
```

[Link to interactive MDS plot](./glimma-plots_1/MDS-Plot.html)

```{r}
#glMDSPlot(lcpm, labels=samplenames, groups = group, launch = T)
```


# Differential expression analysis
```{r echo=TRUE}
design <- model.matrix(~0+group)
colnames(design) <- gsub("group", "", colnames(design))

contr.matrix <- makeContrasts(
  D1vsUntrt = D1-Untrt,
  D2vsUntrt = D2-Untrt,
  D6vsUntrt = D6-Untrt,
  levels = colnames(design)
)

v <- voomWithQualityWeights(x, design, col = col.group, plot=T)
```

```{r echo=TRUE}
fit <- lmFit(v)
fit <- contrasts.fit(fit, contrasts = contr.matrix)
efit <- eBayes(fit)
dt <- decideTests(efit, p.value = 0.05)
summary(dt) #number of DE genes
```

```{r echo=TRUE, fig.height=5, fig.width=5}
##common genes
de.common <- which(dt[,2]!=0 & dt[,3]!=0)
length(de.common)
head(efit$genes$SYMBOL[de.common], n=20)
vennDiagram(dt[,1:3],circle.col = brewer.pal(3,"Set1"))
write.fit(efit,dt, file = "results.txt")
```

[Link to results for three comparisons](./results.txt)


```{r echo=TRUE}
D1.vs.Untrt <- topTable(efit, coef=1,n=Inf)
D2.vs.Untrt <- topTable(efit, coef=2,n=Inf)
D6.vs.Untrt <- topTable(efit, coef=3,n=Inf)
head(D1.vs.Untrt[,c(1,3,4,20,21,22,23,24)])
head(D2.vs.Untrt[,c(1,3,4,20,21,22,23,24)])
head(D6.vs.Untrt[,c(1,3,4,20,21,22,23,24)])

write.table(D1.vs.Untrt, file = "D1_vs_Untrt_RankedResults.txt",sep = "\t",row.names = F)
write.table(D2.vs.Untrt, file = "D2_vs_Untrt_RankedResults.txt",sep = "\t",row.names = F)
write.table(D6.vs.Untrt, file = "D6_vs_Untrt_RankedResults.txt",sep = "\t",row.names = F)
```

[Link to Top Table for Day1 vs Untrt](./D1_vs_Untrt_RankedResults.txt)

[Link to Top Table for Day2 vs Untrt](./D2_vs_Untrt_RankedResults.txt)

[Link to Top Table for Day6 vs Untrt](./D6_vs_Untrt_RankedResults.txt)


```{r}
library(scales)
selgenes = c("Cbfb", "Myh11")
geneind = match(selgenes, fit$genes$SYMBOL)
par(mfrow=c(2,2))

plotMD(efit, coef=1, status=dt[,1], hl.col=alpha(c("red", "blue"),0.6), main="Day1 vs Untreated")
points(efit$Amean[geneind], efit$coef[geneind,1], pch=4, col="orange")
text(efit$Amean[geneind], efit$coef[geneind,1], labels=selgenes, cex=0.75, col="orange")
abline(h=0, col="grey", lty=2)

plotMD(efit, coef=2, status=dt[,2], hl.col=alpha(c("red", "blue"),0.6), main="Day2 vs Untreated")
points(efit$Amean[geneind], efit$coef[geneind,2], pch=4, col="orange")
text(efit$Amean[geneind], efit$coef[geneind,2], labels=selgenes, cex=0.75, col="orange")
abline(h=0, col="grey", lty=2)

plotMD(efit, coef=3, status=dt[,3], hl.col=alpha(c("blue", "red"),0.6), main="Day6 vs Untreated")
points(efit$Amean[geneind], efit$coef[geneind,3], pch=4, col="orange")
text(efit$Amean[geneind], efit$coef[geneind,3], labels=selgenes, cex=0.75, col="orange")
abline(h=0, col="grey", lty=2)
```

[Link to Interactive MDPlot for Day1 vs Untrt](./glimma-plots_1/MD-Plot.html)

[Link to Interactive MDPlot for Day2 vs Untrt](./glimma-plots_2/MD-Plot.html)

[Link to Interactive MDPlot for Day6 vs Untrt](./glimma-plots_3/MD-Plot.html)

```{r eval=F}
glMDPlot(efit, coef=1, status = dt[,1], main = colnames(efit)[1], counts = x$counts,samples=colnames(x), anno = x$genes, groups = group, side.main ="ENTREZID", display.columns = c("SYMBOL","ENTREZID"), search.by="SYMBOL", launch = F)

glMDPlot(efit, coef=2, status = dt[,2], main = colnames(efit)[2], counts = x$counts,samples=colnames(x), anno = x$genes, groups = group, side.main ="ENTREZID", display.columns = c("SYMBOL","ENTREZID"), search.by="SYMBOL", launch = F)

glMDPlot(efit, coef=3, status = dt[,3], main = colnames(efit)[3], counts = x$counts,samples=colnames(x), anno = x$genes, groups = group, side.main ="ENTREZID", display.columns = c("SYMBOL","ENTREZID"), search.by="SYMBOL", launch = F)
```
