<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Yi Xie" />


<title>RNA-seq time course</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>



<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="fluid-row" id="header">



<h1 class="title toc-ignore">RNA-seq time course</h1>
<h4 class="author"><em>Yi Xie</em></h4>
<h4 class="date"><em>19/07/2018</em></h4>

</div>


<div id="background" class="section level1">
<h1>Background</h1>
<p>This experiment is an oncogene withdrawal study where leukaemias are driven by tet transactivator expression of the CBFb-MYH11 fusion (known as inv(16), which is withdrawn upon Dox treatment. In this experiment, we have 3 untreated samples vs 3 samples at both day1, day2 and day6. We got the basic analysis below:</p>
<ol style="list-style-type: decimal">
<li>MD plots of day1 vs untreated, day2 vs untreated and day6 vs untreated.</li>
<li>Lists of DE genes with individual RPKM.</li>
<li>MD plots highlighting Cbfb and Myh11.</li>
</ol>
</div>
<div id="alignmentcounting" class="section level1">
<h1>Alignment&amp;Counting</h1>
<pre class="r"><code>setwd(&quot;/stornext/General/data/user_managed/grpu_mritchie_1/YiXie/reference/mm/&quot;)
library(Rsubread)
library(Mus.musculus)

#build index
buildindex(&#39;Mus_musculus.GRCm38&#39;, reference = &quot;Mus_musculus.GRCm38.dna.primary_assembly.fa&quot;,
           memory = 12000)

#alignment
fastq.files &lt;- list.files(&quot;/stornext/General/data/user_managed/grpu_mritchie_1/MarkMcKenzie/cbf-myh11&quot;, pattern = &quot;fastq.gz$&quot;, full.names=T)
bam &lt;- gsub(&quot;fastq.gz&quot;, &quot;subread.bam&quot;,fastq.files)

align(index = &quot;Mus_musculus.GRCm38&quot;, readfile1 = fastq.files,input_format=&quot;FASTQ&quot;, output_format=&quot;BAM&quot;, 
      output_file=bam, nthreads=8)

#counting&amp;annotation
bam.files &lt;- list.files(path = &quot;./bamfiles&quot;, pattern = &quot;.bam$&quot;, full.names = TRUE)
counts &lt;- featureCounts(bam.files, annot.inbuilt = &quot;mm10&quot;)
names(counts) #see slots in counts
dim(counts$counts) #see the number of genes

geneid &lt;- rownames(counts$counts)
genes &lt;- select(Mus.musculus, keys = geneid, columns = c(&quot;SYMBOL&quot;, &quot;TXCHROM&quot;,&quot;TXSTART&quot;,&quot;TXEND&quot;,&quot;GENENAME&quot;), keytype = &quot;ENTREZID&quot;)
genes &lt;- genes[!duplicated(genes$ENTREZID),]
save(counts,genes,file=&quot;counts.rda&quot;)</code></pre>
</div>
<div id="preprocessing" class="section level1">
<h1>Preprocessing</h1>
<pre class="r"><code>library(edgeR)
library(limma)
library(Glimma)
load(&quot;counts.rda&quot;)

#preprocessing
rpkmvalues &lt;- rpkm(counts$counts, gene.length = counts$annotation$Length)

colnames(rpkmvalues) &lt;- gsub(&quot;X.stornext.General.data.user_managed.grpu_mritchie_1.YiXie.reference.mm.bamfiles.&quot;,&quot;&quot;,colnames(rpkmvalues))
colnames(rpkmvalues) &lt;- gsub(&quot;_S._R1.subread.bam&quot;,&quot;&quot;,colnames(rpkmvalues))
colnames(rpkmvalues) &lt;- samplenames &lt;- gsub(&quot;_S.._R1.subread.bam&quot;,&quot;&quot;,colnames(rpkmvalues))

genes &lt;- cbind(genes, &quot;Length&quot;=counts$annotation$Length, rpkmvalues)

x &lt;- DGEList(counts = counts$counts, genes=genes)

group &lt;- as.factor(c(&quot;D1&quot;,&quot;D1&quot;,&quot;D1&quot;,&quot;D6&quot;,&quot;D6&quot;,&quot;D2&quot;,&quot;D2&quot;,&quot;D2&quot;,&quot;Untrt&quot;,&quot;Untrt&quot;,&quot;Untrt&quot;,&quot;D6&quot;))
x$samples$group &lt;- group
rownames(x$samples) &lt;- gsub(&quot;X.stornext.General.data.user_managed.grpu_mritchie_1.YiXie.reference.mm.&quot;,&quot;&quot;,rownames(x$samples))
x$samples</code></pre>
<pre><code>##                                  group lib.size norm.factors
## bamfiles.1072_S4_R1.subread.bam     D1 47364803            1
## bamfiles.1073_S5_R1.subread.bam     D1 39900483            1
## bamfiles.1074_S6_R1.subread.bam     D1 35506171            1
## bamfiles.1078_S10_R1.subread.bam    D6 30892187            1
## bamfiles.1080_S12_R1.subread.bam    D6 36078384            1
## bamfiles.1081_S7_R1.subread.bam     D2 33655059            1
## bamfiles.1082_S8_R1.subread.bam     D2 49232518            1
## bamfiles.1083_S9_R1.subread.bam     D2 57307065            1
## bamfiles.1084_S3_R1.subread.bam  Untrt 38683405            1
## bamfiles.1086_S2_R1.subread.bam  Untrt 43150122            1
## bamfiles.1087_S1_R1.subread.bam  Untrt 34679913            1
## bamfiles.1979_S11_R1.subread.bam    D6 30775956            1</code></pre>
<pre class="r"><code>##filter lowly expressed
keep.exprs &lt;- rowSums(cpm(x)&gt;1) &gt;=3 #row=genes col=samples; genes express in at least 3 samples
x &lt;- x[keep.exprs,,keep.lib.sizes=F]
dim(counts$counts)</code></pre>
<pre><code>## [1] 27179    12</code></pre>
<pre class="r"><code>dim(x)</code></pre>
<pre><code>## [1] 11675    12</code></pre>
<pre class="r"><code>##normalising
x &lt;- calcNormFactors(x, method = &quot;TMM&quot;)
x$samples$norm.factors</code></pre>
<pre><code>##  [1] 1.0011728 0.9881613 0.9909185 1.0126485 1.0045901 0.9997815 0.9984825
##  [8] 1.0057558 0.9848156 0.9975939 0.9952433 1.0214132</code></pre>
<pre class="r"><code>library(RColorBrewer)
par(mfrow=c(1,1))
col.group &lt;- group
levels(col.group) &lt;- brewer.pal(nlevels(col.group),&quot;Set1&quot;) #change groupID to colorID
col.group &lt;- as.character(col.group)
lcpm &lt;- cpm(x, log=T)

plotMDS(lcpm, labels=group, main=&quot;MDS: Sample Groups&quot;, col=col.group)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-5-1.png" width="480" style="display: block; margin: auto;" /></p>
<p><a href="./glimma-plots_1/MDS-Plot.html">Link to interactive MDS plot</a></p>
<pre class="r"><code>#glMDSPlot(lcpm, labels=samplenames, groups = group, launch = T)</code></pre>
</div>
<div id="differential-expression-analysis" class="section level1">
<h1>Differential expression analysis</h1>
<pre class="r"><code>design &lt;- model.matrix(~0+group)
colnames(design) &lt;- gsub(&quot;group&quot;, &quot;&quot;, colnames(design))

contr.matrix &lt;- makeContrasts(
  D1vsUntrt = D1-Untrt,
  D2vsUntrt = D2-Untrt,
  D6vsUntrt = D6-Untrt,
  levels = colnames(design)
)

v &lt;- voomWithQualityWeights(x, design, col = col.group, plot=T)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
<pre class="r"><code>fit &lt;- lmFit(v)
fit &lt;- contrasts.fit(fit, contrasts = contr.matrix)
efit &lt;- eBayes(fit)
dt &lt;- decideTests(efit, p.value = 0.05)
summary(dt) #number of DE genes</code></pre>
<pre><code>##        D1vsUntrt D2vsUntrt D6vsUntrt
## Down         211       345      1572
## NotSig     11227     10898      8554
## Up           237       432      1549</code></pre>
<pre class="r"><code>##common genes
de.common &lt;- which(dt[,2]!=0 &amp; dt[,3]!=0)
length(de.common)</code></pre>
<pre><code>## [1] 679</code></pre>
<pre class="r"><code>head(efit$genes$SYMBOL[de.common], n=20)</code></pre>
<pre><code>##  [1] &quot;Mrpl15&quot;  &quot;Lypla1&quot;  &quot;Ncoa2&quot;   &quot;Mcm3&quot;    &quot;Smap1&quot;   &quot;Arhgef4&quot; &quot;Neurl3&quot; 
##  [8] &quot;Cox5b&quot;   &quot;Aff3&quot;    &quot;Rnf149&quot;  &quot;Il1rl1&quot;  &quot;Ormdl1&quot;  &quot;Coq10b&quot;  &quot;Ndufb3&quot; 
## [15] &quot;Als2&quot;    &quot;Rpe&quot;     &quot;Rufy4&quot;   &quot;Chpf&quot;    &quot;Sp110&quot;   &quot;Ube2f&quot;</code></pre>
<pre class="r"><code>vennDiagram(dt[,1:3],circle.col = brewer.pal(3,&quot;Set1&quot;))</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-9-1.png" width="480" /></p>
<pre class="r"><code>write.fit(efit,dt, file = &quot;results.txt&quot;)</code></pre>
<p><a href="./results.txt">Link to results for three comparisons</a></p>
<pre class="r"><code>D1.vs.Untrt &lt;- topTable(efit, coef=1,n=Inf)
D2.vs.Untrt &lt;- topTable(efit, coef=2,n=Inf)
D6.vs.Untrt &lt;- topTable(efit, coef=3,n=Inf)
head(D1.vs.Untrt[,c(1,3,4,20,21,22,23,24)])</code></pre>
<pre><code>##       ENTREZID  SYMBOL TXCHROM      logFC  AveExpr         t      P.Value
## 56508    56508 Rapgef4    chr2 -2.0873337 4.638095 -28.32372 8.375169e-13
## 14866    14866   Gstm5    chr3 -1.3747527 6.550460 -16.50372 6.425772e-10
## 71279    71279 Slc29a3   chr10 -0.9342152 5.911305 -13.50696 7.098753e-09
## 12400    12400    Cbfb    chr8 -0.5804319 8.166644 -11.83651 3.352621e-08
## 66124    66124   Josd2    chr7 -1.1568932 6.934212 -11.11436 6.954805e-08
## 74277    74277   Chic2    chr5 -0.7300277 6.394060 -10.81890 9.481896e-08
##          adj.P.Val
## 56508 9.778010e-09
## 14866 1.875522e-06
## 71279 1.657559e-05
## 12400 6.523641e-05
## 66124 1.159962e-04
## 74277 1.383764e-04</code></pre>
<pre class="r"><code>head(D2.vs.Untrt[,c(1,3,4,20,21,22,23,24)])</code></pre>
<pre><code>##       ENTREZID  SYMBOL TXCHROM      logFC  AveExpr         t      P.Value
## 56508    56508 Rapgef4    chr2 -2.0576175 4.638095 -27.13062 1.430064e-12
## 53313    53313  Atp2a3   chr11  0.8730780 8.688141  17.73569 2.682222e-10
## 14866    14866   Gstm5    chr3 -1.4826494 6.550460 -17.04082 4.359080e-10
## 71279    71279 Slc29a3   chr10 -1.0719108 5.911305 -14.80930 2.368847e-09
## 66124    66124   Josd2    chr7 -1.4458910 6.934212 -13.20350 9.291183e-09
## 74277    74277   Chic2    chr5 -0.8294331 6.394060 -11.79216 3.502425e-08
##          adj.P.Val
## 56508 1.669599e-08
## 53313 1.017845e-06
## 14866 1.017845e-06
## 71279 4.609382e-06
## 66124 1.549637e-05
## 74277 5.111351e-05</code></pre>
<pre class="r"><code>head(D6.vs.Untrt[,c(1,3,4,20,21,22,23,24)])</code></pre>
<pre><code>##       ENTREZID  SYMBOL TXCHROM     logFC  AveExpr         t      P.Value
## 56508    56508 Rapgef4    chr2 -2.219838 4.638095 -28.53749 7.627214e-13
## 53313    53313  Atp2a3   chr11  1.095229 8.688141  23.34446 9.209207e-12
## 14866    14866   Gstm5    chr3 -1.650777 6.550460 -19.35708 9.221930e-11
## 27027    27027 Tspan32    chr7  1.800441 4.872252  17.85462 2.472677e-10
## 74772    74772 Atp13a2    chr4  1.087912 7.105460  17.32844 3.557542e-10
## 71279    71279 Slc29a3   chr10 -1.217692 5.911305 -17.03795 4.368019e-10
##          adj.P.Val
## 56508 8.904772e-09
## 53313 5.375875e-08
## 14866 2.691651e-07
## 27027 4.811417e-07
## 74772 5.933472e-07
## 71279 6.374578e-07</code></pre>
<pre class="r"><code>write.table(D1.vs.Untrt, file = &quot;D1_vs_Untrt_RankedResults.txt&quot;,sep = &quot;\t&quot;,row.names = F)
write.table(D2.vs.Untrt, file = &quot;D2_vs_Untrt_RankedResults.txt&quot;,sep = &quot;\t&quot;,row.names = F)
write.table(D6.vs.Untrt, file = &quot;D6_vs_Untrt_RankedResults.txt&quot;,sep = &quot;\t&quot;,row.names = F)</code></pre>
<p><a href="./D1_vs_Untrt_RankedResults.txt">Link to Top Table for Day1 vs Untrt</a></p>
<p><a href="./D2_vs_Untrt_RankedResults.txt">Link to Top Table for Day2 vs Untrt</a></p>
<p><a href="./D6_vs_Untrt_RankedResults.txt">Link to Top Table for Day6 vs Untrt</a></p>
<pre class="r"><code>library(scales)
selgenes = c(&quot;Cbfb&quot;, &quot;Myh11&quot;)
geneind = match(selgenes, fit$genes$SYMBOL)
par(mfrow=c(2,2))

plotMD(efit, coef=1, status=dt[,1], hl.col=alpha(c(&quot;red&quot;, &quot;blue&quot;),0.6), main=&quot;Day1 vs Untreated&quot;)
points(efit$Amean[geneind], efit$coef[geneind,1], pch=4, col=&quot;orange&quot;)
text(efit$Amean[geneind], efit$coef[geneind,1], labels=selgenes, cex=0.75, col=&quot;orange&quot;)
abline(h=0, col=&quot;grey&quot;, lty=2)

plotMD(efit, coef=2, status=dt[,2], hl.col=alpha(c(&quot;red&quot;, &quot;blue&quot;),0.6), main=&quot;Day2 vs Untreated&quot;)
points(efit$Amean[geneind], efit$coef[geneind,2], pch=4, col=&quot;orange&quot;)
text(efit$Amean[geneind], efit$coef[geneind,2], labels=selgenes, cex=0.75, col=&quot;orange&quot;)
abline(h=0, col=&quot;grey&quot;, lty=2)

plotMD(efit, coef=3, status=dt[,3], hl.col=alpha(c(&quot;blue&quot;, &quot;red&quot;),0.6), main=&quot;Day6 vs Untreated&quot;)
points(efit$Amean[geneind], efit$coef[geneind,3], pch=4, col=&quot;orange&quot;)
text(efit$Amean[geneind], efit$coef[geneind,3], labels=selgenes, cex=0.75, col=&quot;orange&quot;)
abline(h=0, col=&quot;grey&quot;, lty=2)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-11-1.png" width="672" /></p>
<p><a href="./glimma-plots_1/MD-Plot.html">Link to Interactive MDPlot for Day1 vs Untrt</a></p>
<p><a href="./glimma-plots_2/MD-Plot.html">Link to Interactive MDPlot for Day2 vs Untrt</a></p>
<p><a href="./glimma-plots_3/MD-Plot.html">Link to Interactive MDPlot for Day6 vs Untrt</a></p>
<pre class="r"><code>glMDPlot(efit, coef=1, status = dt[,1], main = colnames(efit)[1], counts = x$counts,samples=colnames(x), anno = x$genes, groups = group, side.main =&quot;ENTREZID&quot;, display.columns = c(&quot;SYMBOL&quot;,&quot;ENTREZID&quot;), search.by=&quot;SYMBOL&quot;, launch = F)

glMDPlot(efit, coef=2, status = dt[,2], main = colnames(efit)[2], counts = x$counts,samples=colnames(x), anno = x$genes, groups = group, side.main =&quot;ENTREZID&quot;, display.columns = c(&quot;SYMBOL&quot;,&quot;ENTREZID&quot;), search.by=&quot;SYMBOL&quot;, launch = F)

glMDPlot(efit, coef=3, status = dt[,3], main = colnames(efit)[3], counts = x$counts,samples=colnames(x), anno = x$genes, groups = group, side.main =&quot;ENTREZID&quot;, display.columns = c(&quot;SYMBOL&quot;,&quot;ENTREZID&quot;), search.by=&quot;SYMBOL&quot;, launch = F)</code></pre>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
